src/tests/math/test_math_arg.f90

```diff
@@ -0,0 +1,41 @@
+program tester
+
+    real :: tol = 1.0e-3
+    call test_math_arg_complex
+    call test_math_argd_complex
+    call test_math_argpi_complex
+    print *, "All tests in `test_math_arg` passed."
+
+contains
+
+    subroutine test_math_arg_complex
+        use stdlib_math, only: arg
+        use stdlib_error, only: check
+
+        call check(abs(arg(2.0*exp((0.0e0, 0.5))) - 0.5) < tol, msg="arg(2.0*exp((0.0, 0.5))) failed.")
```

Comment:
A general comment on the tests here for all variants of these routines. 

Even though this routine is simple, I think it would be good to make "tighter" tests, for example by confirming that (for `argpi`) `z = abs(z)*exp((0.0, theta*π))` is true to a high tolerance, likely higher than 1e-03 (perhaps some multiple of `epsilon(1.0_theprecision)` 

It would also be good to check a greater range of complex numbers (in different quadrants, and considering corner cases such as the discontinuity in the degrees function). 

I also think there should be tests for other supported real kinds, with appropriate tolerances. You could auto-generate these with fypp, see example in the tests for the statistics module.

Reply:
Thanks for your advice @gareth-nx . 
I used a smaller tolerance `epsilon (1.0_sp)` to test `arg/argd/argpi`, it seems that it is not very necessary to use fypp to test `kind=dp/qp`?